presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java

```diff
@@ -769,6 +736,41 @@ public void revokeTablePrivileges(String databaseName, String tableName, String
         }
     }
 
+    private PrivilegeBag buildPrivilegeBag(HiveMetastoreClient metastoreClient,
+                                                     String databaseName,
+                                                     String tableName,
+                                                     String grantee,
+                                                     Set<PrivilegeGrantInfo> privilegeGrantInfos)
+    {
+        PrincipalType principalType;
+        String principalName;
+
+        try {
+            if (metastoreClient.getRoleNames().contains(grantee.toLowerCase())) {
+                principalType = ROLE;
+                principalName = grantee.toLowerCase(); // Hive metastore API requires role names to be in lowercase.
+            }
+            else {
+                principalType = USER;
+                principalName = grantee;
+            }
+
+            ImmutableList.Builder<HiveObjectPrivilege> privilegeBagBuilder = ImmutableList.builder();
+            for (PrivilegeGrantInfo privilegeGrantInfo : privilegeGrantInfos) {
+                privilegeBagBuilder.add(
+                        new HiveObjectPrivilege(
+                                new HiveObjectRef(HiveObjectType.TABLE, databaseName, tableName, null, null),
+                                principalName,
+                                principalType,
+                                privilegeGrantInfo));
+            }
+            return new PrivilegeBag(privilegeBagBuilder.build());
+        }
+        catch (TException e) {
```

Comment:
I think you want to remove the exception handling from this method, so the retry handler can process the raw exception